### PR TITLE
Small belt overlay rewrite, and tool overlays for toolbelts

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis & MoreRobustThanYou
imageadd: Toolbelts now have overlays for crowbars, wirecutters, screwdrivers, multitools, and wrenches.
/:cl:

Credit to @MoreRobustThanYou for the idea and initial concept.

Closes #29160.

I saw #29160, and figured that the idea was worth pursuing, but the sprites themselves simply looked too bad for it to be worth adding. To get around this, I sprited overlays for the crowbar, wirecutter, screwdriver, multitool, and wrench (and all their color variants, plus the brass ones) on the toolbelt. I opted to take the security belt route of only showing some items, as the toolbelt isn't large enough to fit every tool without looking really cluttered.

I also realized that overlays were handled based on proper name. To address this, the overlays now use the item's icon state. Additionally, belts now have a `content_overlays` var; overlays of objects will only appear on the belt if this is set to `TRUE`.

Finally, belt overlays now have their own file, at `belt_overlays.dmi`.

### CODE REVIEW

**Comment by Armhulen** [comment]
i just don't like it!!!

**Comment by ChangelingRain** [comment]
bleh bleh these aren't dynamic based on item sprites so I really don't like it

**Comment by lzimann** [comment]
It really doesn't look good, plus, an extra sprite everytime someone wants to add an extra tool. Just not worth it.

**Comment by Xhuis** [comment]
I'm not sure it's actually possible to generate dynamic sprites, and if it is, I'm a doofus and don't know about it. Ultimately, it's up to y'all if the pretty aspect is worth keeping.

**Comment by Xhuis** [comment]
Point of note: if you're _adamantly_ against it and would close it, let me know so I can remove the bad parts and keep the code improvements instead of just closing the PR and losing everything.

**Comment by Mark9013100** [comment]
The sprite's way too busy, I could see something like the wrench display akin to the baton on the sec belt though.

**Comment by MoreRobustThanYou** [comment]
@Xhuis I didn't come up with the idea, I ripped the sprites STRAIGHT from Paradise

**Comment by optimumtact** [comment]
Is it just me or does the sprite look worse than the original belt sprite?

**Comment by Xhuis** [comment]
What parts can be improved?

**Comment by Kraseo** [comment]
why have brass sprites when cogscarabs get them in a box

**Comment by Xhuis** [comment]
Completionism, in case someone gets a hold of them somehow

**Comment by MoreRobustThanYou** [comment]
@Xhuis crazy idea: dynamically make textures at compile by shrinking the item sprites to the size you need for the belt

**Comment by Xhuis** [comment]
I'm not sure how to do that.

**Comment by MoreRobustThanYou** [comment]
neither do I

**Review by lzimann — DISMISSED** [review]
It should either generate dynamically based on already existing sprites or not happen at all. This kind of thing is awful to maintain and should really be avoided.

**Comment by Xhuis** [comment]
If you know how to generate dynamic sprites like that, then it's the solution I'd prefer as well. I'm just not sure how to implement it. Perhaps with `mutable_appearance`s? I don't know how to work with those, but I can easily figure it out.

**Comment by WJohn** [comment]
These are much better. Good work.

**Comment by lzimann** [comment]
Adding more on this: 
You are adding 22 new sprites on the BELT icon file, which by itself is already a maintainance hinder.
You didn't even make everything that can be put inside a toolbelt, no radios, no t-rays, no flashlights, no CE tools, doing each would require even more sprites.

So no, this will not happen like this.

**Comment by Xhuis** [comment]
Like I mentioned in the OP, I didn't make sprites for everything intentionally, just like the sec belt doesn't have everything. This is also why I'm not overly concerned about people claiming that maintaining it will be diffiicult if a new tool is added, because there's no reason to add overlays for every tools. Also like I said, I would do a better way if I knew how, but I have so far been left in the dark about how.

Overlays can also easily be put into their own .dmi.

I'm doing everything I can here.

**Comment by ChangelingRain** [comment]
Frankly unless this is a dynamic thing I'm inclined to deny this, since, despite being interesting, it requires too many new sprites if someone wants to add a new tool to it.

**Comment by ChangelingRain** [comment]
It's not dynamic if someone has to make a sprite for it.

**Comment by Xhuis** [comment]
"Dynamic" in this case is a word being thrown around a lot. It's dynamic in some ways, but not in others. Generating a sprite based on appearance would be _possible_, but being able to make ones to your own liking can make it look a lot better - if you take a look at the other PR, shoddily done generated overlays can end up looking horrific.

That said, I'm about to push a commit that moves the overlays to another file, and then I'm gonna try to mess around with appearances to generate overlays myself. This might still be costly - I'd have to get a position for each tool as well as a size and rotation - but it's what everyone views as a better solution so I'm willing to try it.

File: code/game/objects/items/weapons/storage/belt.dm [hunk 1]
```diff
@@ -7,13 +7,19 @@
...
 	max_integrity = 300
+	var/content_overlays = FALSE //If this is true, the belt will gain overlays based on what it's holding
 
 /obj/item/weapon/storage/belt/update_icon()
 	cut_overlays()
-	for(var/obj/item/I in contents)
-		add_overlay("[I.name]")
+	if(content_overlays)
+		for(var/obj/item/I in contents)
+			add_overlay(icon('icons/obj/clothing/belt_overlays.dmi', I.icon_state))
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/storage/belt.dm]: no.
this should be a ```mutable_appearance```.

What even went through your head to make you think ```icon()``` was acceptable here?

**Review by RemieRichards — CHANGES_REQUESTED** [review]
Bad overlay code.

File: code/game/objects/items/weapons/storage/belt.dm (hunk 1 above)
Comment (Xhuis) [inline code/game/objects/items/weapons/storage/belt.dm]: It was probably the fact that humans make mistakes.

**Comment by Xhuis** [comment]
I'm going to be working on this at some point soonish, but there's other stuff I want to get done first. This might be stale for a few hours.

**Comment by RemieRichards** [comment]
@MrStonedOne do you think the CDN will allow us to use more icons for things?
ie: the hair extension system for hats (already in), this, etc?

Since while the downloads are bigger, they are more local so they should be faster (in theory)

**Comment by Ausops** [comment]
one of these days inhand sprites should be turned into the mini-version of an item sprite and dynamically edited for use in inhand directions but also things like this

**Comment by RemieRichards** [comment]
Inhands will never look good as shrunken versions of the main sprites, and as such we won't ever go there.

**Comment by Ausops** [comment]
badly done inhands you mean

**Comment by Bawhoppen** [comment]
Add overlays to the grenade belt please, I always meant to and the sprite is even made for it but I never added them
Thank you

**Comment by MrStonedOne** [comment]
>Since while the downloads are bigger, they are more local so they should be faster (in theory)

The downloads being bigger was fixed, the resource file it serves up is identical now to the resource file served up by the game server.

**Comment by RemieRichards** [comment]
@MrStonedOne uh, no, my point wasn't about some bug with the CDN setup.
it's the fact that if we were to go this route, there'd be infinitely more sprites, thus, larger downloads.

**Comment by MrStonedOne** [comment]
oh, ya, it should be fine now to do much larger rscs

**Comment by Xhuis** [comment]
I'd like to be able to finish this up, but with the greyscale screwdriver sprite I'm unsure if it's really doable in an effective way. I'd ideally have the greyscale color _and_ the head, but without an `istype()` check for every object added to the belt (hacky) it doesn't seem to be possible.

Are there any existing solutions to this problem, before I make one myself?

**Comment by MrStonedOne** [comment]
I need more info.

**Comment by Xhuis** [comment]
I'm not sure if you missed the screwdriver PR, so here's the short version:

Basically, screwdrivers' icons are now made through the greyscale handle being the actual `icon_state`, and the screwdriver's head and highlights are added afterwards. This is so that the handle can be any color, instead of the seven it used to be.

However, therein lies the current problem. Overlays for belts, as handled right now, use the object's `icon_state`. The issue being, here, the screwdriver on the belt is greyscale. That's all fine - we can color it - but it _doesn't_ account for the head and highlights, which _shouldn't_ be colored. If I keep them in the sprite, they would appear whatever color the rest of the screwdriver is, when I only want the handle to be colored.

**Comment by ChangelingRain** [comment]
Could always make screwdrivers static again.

**Comment by RemieRichards** [comment]
@Xhuis 
```
/obj/item/proc/get_belt_overlay()
    return mutable_appearance('belt_overlays.dmi', icon_state)

/obj/item/weapon/screwdriver/get_belt_overlay()
    var/mutable_appearance/head = mutable_appearance('belt_overlays', "screw_head")
    var/mutable_appearance/body = mutable_appearance('belt_overlays', "screw_body_greyscale")
    body.color = "some colour here"
    head.overlays +=body
    return head
``` 

?

**Comment by MrStonedOne** [comment]
Overlays can have overlays themselves.

You could do the object's appearance as the overlay, this would add the objects overlays as well, or make the overlay as an /image/ and add the object's overlays if you have to do some pre-processing.

**Comment by RemieRichards** [comment]
@MrStonedOne look one comment above you buddy boy.

**Comment by MrStonedOne** [comment]
That comment was there when I posted my comment blame github for having a shitty auto-refresh

**Comment by RemieRichards** [comment]
It's usually not that bad!

**Comment by lzimann** [comment]
I dislike this and has been stale for like 6 days already. Someone can reopen if they care about this.

**Comment by RemieRichards** [comment]
The only valid, actionable thing about that statement is 'it's been stale for 6 days'

**Comment by lzimann** [comment]
like i said, someone can reopen if they care/this keeps going

**Comment by Xhuis** [comment]
Well, I _was_ going to do some work on this, but given that I imagine nobody feels like reopening it, uhh, rest in peace, PR.

**Comment by RemieRichards** [comment]
You don't know that.
Look at what I said above, I've only not re-opened it because it *IS* stale.

**Comment by Xhuis** [comment]
Oh! I thought you were against it. Well, I'm not sure if pushing commits to closed PRs does anything, but I'll try that anyway.

**Comment by RemieRichards** [comment]
it doesn't display unless someone reopens it, but I'll take this as a contract that you'll finish it.

**Comment by Xhuis** [comment]
I'll _try_ to finish it, and assuming I can get this to work I definitely will. If I can't end up doing it, though, I'll close it myself!

**Comment by lzimann** [comment]
I'll be honest though, I still think this is needless bullshit that will be a massive pain to keep.

**Comment by Xhuis** [comment]
I won't bash you for thinking that. We've had sec belt overlays for a really long time, though, and the solution it used is just by only having overlays for a few different objects, not all of them. I don't see why toolbelts can't be the same.

**Comment by Xhuis** [comment]
Through @RemieRichards's help I've gotten screwdriver overlays working. The handle greyscales correctly, the head does not, and any overridden screwdrivers show the right sprite.

Assuming that was the only thing keeping this back, it should be ready now.

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -181,6 +181,16 @@
...
 
+/obj/item/weapon/screwdriver/get_belt_overlay()
+	if(random_color)
+		var/mutable_appearance/body = mutable_appearance('icons/obj/clothing/belt_overlays.dmi', "screwdriver")
+		var/mutable_appearance/head = mutable_appearance('icons/obj/clothing/belt_overlays.dmi', "screwdriver_head")
+		body.color = color
+		head.overlays += body
+		return head
+	else
+		return mutable_appearance('icons/obj/clothing/belt_overlays.dmi', item_state) //Item state, because some screwdrivers use different icon files
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/tools.dm]: @Xhuis explain this more please.

Comment (Xhuis) [inline code/game/objects/items/weapons/tools.dm]: The abductor screwdriver refers to `abductor.dmi` to get its icon state. It uses the `item_state` of a regular screwdriver, however - `screwdriver_nuke` - and so I refer to it there.

I can also change its icon state directly if that's more consistent.

Comment (RemieRichards) [inline code/game/objects/items/weapons/tools.dm]: please do, consistency is nice

**Review by RemieRichards — APPROVED** [review]

**Comment by Xhuis** [comment]
\o/ thank you!

**Pull request merged**